src/cls/rbd/cls_rbd_types.h

```diff
@@ -248,13 +253,13 @@ struct GroupSnapshotNamespace {
 
   GroupSnapshotNamespace(int64_t _group_pool,
 			 const string &_group_id,
-			 const snapid_t &_snapshot_id) :group_pool(_group_pool),
-							group_id(_group_id),
-							snapshot_id(_snapshot_id) {}
+			 const string &_snapshot_id) :group_pool(_group_pool),
+						      group_id(_group_id),
+						      snapshot_id(_snapshot_id) {}
 
   int64_t group_pool;
   string group_id;
-  snapid_t snapshot_id;
+  string snapshot_id;
```

Comment:
After trying to rename it I realized that it looks confusing from user's perspective.
If I have code in librbd/internal.cc like:
```
rados.ioctx_create2(snap_namespace.group_pool, group_ioctx);
```
or 
```
string group_header_oid = util::group_header_name(snap_namespace.group_id);
```
When it only has id instead of group_id then the impression is that it's a namespace id rather then a group id that this namespace references. In this regard I would prefer to add group to snapshot_id.